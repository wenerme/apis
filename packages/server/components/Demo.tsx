import React from 'react'

export const Demo: React.FC = () => {
  return (
    <div>

    </div>
  )
};

