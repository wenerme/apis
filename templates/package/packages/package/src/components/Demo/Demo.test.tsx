import React from 'react';

import renderer, { act } from 'react-test-renderer';
import { Demo } from 'src/components/Demo/Demo';

test('Demo component test', async () => {
  const component = renderer.create(<Demo title="Title">content</Demo>);

  let tree = component.toJSON() as any;
  expect(tree).toMatchSnapshot();

  // manually trigger the callback
  act(() => tree.children.find(v => v.type === 'h2').props.onMouseEnter());
  // re-rendering
  tree = component.toJSON();
  expect(tree).toMatchSnapshot();

  // manually trigger the callback
  act(() => tree.children.find(v => v.type === 'h2').props.onMouseLeave());
  // re-rendering
  tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
