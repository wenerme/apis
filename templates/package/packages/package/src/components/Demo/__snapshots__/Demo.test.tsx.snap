// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Demo component test 1`] = `
<div>
  <h2
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    Demo 
    Title
     
  </h2>
  <div>
    content
  </div>
</div>
`;

exports[`Demo component test 2`] = `
<div>
  <h2
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    Demo 
    Title
     
    <a
      href="#"
    >
      #
    </a>
  </h2>
  <div>
    content
  </div>
</div>
`;

exports[`Demo component test 3`] = `
<div>
  <h2
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    Demo 
    Title
     
  </h2>
  <div>
    content
  </div>
</div>
`;
